models/location_helpers.go

```diff
@@ -104,6 +103,39 @@ func checkS3Config(c *S3LocationConfig) error {
 	return nil
 }
 
+// checkS3Config checks S3 config and returns the flag that indicates if
+// secure connection should be used and the parsed host.
+func checkS3Config(c *S3LocationConfig) (bool, string, error) {
+	if err := s3ConfigFilled(c); err != nil {
+		return false, "", err
+	}
+
+	parsedURL, err := url.Parse(c.Endpoint)
+	if err != nil {
+		return false, "", status.Errorf(codes.InvalidArgument, "%s", err)
+	}
+
+	if parsedURL.Host == "" {
+		return false, "", status.Error(codes.InvalidArgument, "No host found in the Endpoint.")
+	}
+
+	if parsedURL.Path != "" && parsedURL.Path != "/" {
+		return false, "", status.Error(codes.InvalidArgument, "Path is not allowed for Endpoint.")
+	}
+
+	secure := true
+	switch parsedURL.Scheme {
+	case "http":
+		secure = false
+	case "https":
+	case "":
+	default:
+		return false, "", status.Errorf(codes.InvalidArgument, "Invalid scheme '%s'", parsedURL.Scheme)
+	}
+
+	return secure, parsedURL.Host, nil
```

Comment:
is it possible that url contains non 443 port?

Reply:
yes, the user can choose desired port for a self hosted minio server.